rosetta/text/text_processors.py

```diff
@@ -695,50 +703,41 @@ def filter_sfile(
                 "call: self.compactify() then either self.set_id2token() or "
                 " self.save() before filtering")
 
-        extra_filter = self._get_extra_filter(doc_id_list)
+        if filters is None:
+            filters = []
 
```

Comment:
Can you add a quick comment here about why you use this order of filters?  It will help maintainers of the code understand the reasoning (your reasoning looks correct).